### PR TITLE
Add product price validation

### LINKED ISSUE #5326: It should not be possible to set negative prices
### Steps to reproduce the problem
1. Go to dashboard 
2. Open any product
3. Set its price to -1

### What I expected to happen
To throw an error

### PR DESCRIPTION
**Pull request by kswiatek92**
closes #5326

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: tests/api/test_product.py [hunk 1]
```diff
@@ -1595,6 +1595,46 @@ def test_update_product_slug_and_name(
...
+    response = staff_api_client.post_graphql(
+        UPDATE_PRODUCT_PRICE_MUTATION,
+        variables,
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["productUpdate"]
+    errors = data["productErrors"]
+    assert errors[0]["field"] == "basePrice"
+    assert errors[0]["code"] == "INVALID"
```
Comment (maarcingebala) [inline tests/api/test_product.py]: We shouldn't hardcode enum values:
```suggestion
    assert errors[0]["code"] == ProductErrorCode.INVALID.name
```

File: saleor/graphql/product/mutations/products.py [hunk 2]
```diff
@@ -833,6 +833,15 @@ def clean_input(cls, info, instance, data):
         # from the schema, only "basePrice" should be used here.
         price = data.get("base_price", data.get("price"))
         if price is not None:
+            if price < 0:
```
Comment (fowczarek) [inline saleor/graphql/product/mutations/products.py]: Should 0 be the correct price in Saleor? 
@patrys @maarcingebala

Comment (NyanKiyoshi) [reply saleor/graphql/product/mutations/products.py]: Definitely, it is common to set a product to free

Comment (fowczarek) [reply saleor/graphql/product/mutations/products.py]: But in most popular case free products have restrictions on the number of items. Currently, we are not able to set it on the product variant. But we can create Voucher\Sale with the price set to 0 and quantity restriction.

Comment (patrys) [reply saleor/graphql/product/mutations/products.py]: Free should be supported, it's not likely for someone to use it for a physical shippable item.

Comment (maarcingebala) [reply saleor/graphql/product/mutations/products.py]: We should be able to set 0 as a product or variant price.

**Comment by maarcingebala** [comment]
@kswiatek92 There are a few more places where we can set prices in API.

We have `costPrice` and `priceOverride` fields in:
- `productVariantUpdate`
- `productVariantCreate`
- `productVariantBulkCreate`

We should ensure that we have validation there as well. If it makes sense maybe we should add a function that does this validation and raises this error so don't have to duplicate entire ValidationError multiple times.

**Comment by kswiatek92** [comment]
@maarcingebala I don't think it's necessary, 
there are only 3 places where this validation occurs, for different fields. To make it work properly, such function should have additional argument for field, and maybe message error, which make such function just ValidationError with different name

**Comment by fowczarek** [comment]
@kswiatek92 
You can create function without restriction that you write. 
As a function argument, you can pass price instead of mutation input. And function can return validation error with the message and you should catch this error in mutation body and add required ErrorCode.

@maarcingebala What do you think about introducing a specific error code for this validation?

File: CHANGELOG.md [hunk 3]
```diff
@@ -86,6 +86,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add ability to assign multiple warehouses in mutations to create/update a shipping zone - #5399 by @fowczarek
 - Add filter by ids to warehouses query - #5414 by @fowczarek
```
Comment (fowczarek) [inline CHANGELOG.md:88]: Please remove it. :policeman:

File: saleor/graphql/product/bulk_mutations/products.py [hunk 4]
```diff
@@ -155,10 +155,28 @@ def clean_input(cls, info, instance: models.ProductVariant, data: dict):
 
         cost_price_amount = cleaned_input.pop("cost_price", None)
         if cost_price_amount is not None:
+            if cost_price_amount < 0:
+                raise ValidationError(
+                    {
+                        "basePrice": ValidationError(
+                            ("Product price cannot be lower than 0."),
```
Comment (fowczarek) [inline saleor/graphql/product/bulk_mutations/products.py:162]:
```suggestion
                            "Product price cannot be lower than 0.",
```

File: saleor/graphql/product/bulk_mutations/products.py [hunk 5]
```diff
@@ -155,10 +155,28 @@ def clean_input(cls, info, instance: models.ProductVariant, data: dict):
...
+                )
             cleaned_input["cost_price_amount"] = cost_price_amount
 
         price_override_amount = cleaned_input.pop("price_override", None)
         if price_override_amount is not None:
+            if price_override_amount < 0:
+                raise ValidationError(
+                    {
+                        "basePrice": ValidationError(
+                            ("Product price cannot be lower than 0."),
```
Comment (fowczarek) [inline saleor/graphql/product/bulk_mutations/products.py:175]: Same.

File: saleor/graphql/product/mutations/products.py [hunk 6]
```diff
@@ -833,6 +833,15 @@ def clean_input(cls, info, instance, data):
...
+                raise ValidationError(
+                    {
+                        "basePrice": ValidationError(
+                            ("Product base price cannot be lower than 0."),
```
Comment (fowczarek) [inline saleor/graphql/product/mutations/products.py:840]: Same.

File: saleor/graphql/product/mutations/products.py [hunk 7]
```diff
@@ -1175,10 +1184,30 @@ def clean_input(
         cleaned_input = super().clean_input(info, instance, data)
 
         if "cost_price" in cleaned_input:
-            cleaned_input["cost_price_amount"] = cleaned_input.pop("cost_price")
+            cost_price = cleaned_input.pop("cost_price")
+            if cost_price < 0:
+                raise ValidationError(
+                    {
+                        "costPrice": ValidationError(
+                            ("Product price cannot be lower than 0."),
```
Comment (fowczarek) [inline saleor/graphql/product/mutations/products.py:1192]: Same.

File: saleor/graphql/product/mutations/products.py [hunk 8]
```diff
@@ -1175,10 +1184,30 @@ def clean_input(
...
+            cleaned_input["cost_price_amount"] = cost_price
 
         if "price_override" in cleaned_input:
-            cleaned_input["price_override_amount"] = cleaned_input.pop("price_override")
+            price_override = cleaned_input.pop("price_override")
+            if price_override < 0:
+                raise ValidationError(
+                    {
+                        "priceOverride": ValidationError(
+                            ("Product price cannot be lower than 0."),
```
Comment (fowczarek) [inline saleor/graphql/product/mutations/products.py:1205]: Same.

File: saleor/graphql/product/mutations/products.py [hunk 9]
```diff
@@ -1175,10 +1184,30 @@ def clean_input(
...
+                            code=ProductErrorCode.INVALID,
+                        )
+                    }
+                )
+            cleaned_input["cost_price_amount"] = cost_price
 
         if "price_override" in cleaned_input:
-            cleaned_input["price_override_amount"] = cleaned_input.pop("price_override")
+            price_override = cleaned_input.pop("price_override")
+            if price_override < 0:
```
Comment (fowczarek) [inline saleor/graphql/product/mutations/products.py:1201]: `ProductVariantUpdate` inherit form this mutation. There is a change that `price_override` in `None`. Did you check it?

**Pull request merged**